sympy/series/sequences.py

```diff
@@ -0,0 +1,1123 @@
+        >>> SeqFormula(n**2).add(SeqFormula(n**3))
+        SeqFormula((n**3 + n**2, n), ([0, oo), 1))
+        """
+        return SeqAdd(self, other)
+
+    def sub(self, other):
+        """
+        Returns the term-wise subtraction of 'self' and 'other'.
+        ``other`` should be a sequence.
+
+        Examples
+        ========
+
+        As a shortcut it is possible to use the '-' operator:
+
+        >>> from sympy import S, oo, SeqFormula
+        >>> from sympy.abc import n
+        >>> SeqFormula(n**2).sub(SeqFormula(n))
+        SeqFormula((n**2 - n, n), ([0, oo), 1))
+        """
+        return self.add(-other)
+
+    def mul(self, other):
+        """
+        Returns the term-wise multiplication of 'self' and 'other'.
+        ``other`` should be a sequence. For ``other`` not being a
+        sequence see ``coeff_mul`` method.
+
+        Examples
+        ========
+
+        As a shortcut it is possible to use the '*' operator:
+
+        >>> from sympy import S, oo, SeqFormula
+        >>> from sympy.abc import n
+
+        >>> SeqFormula(n**2).mul(SeqFormula(n))
+        SeqFormula((n**3, n), ([0, oo), 1))
+        """
+        if isinstance(other, SeqBase):
+            return SeqMul(self, other)
+
+    def coeff_mul(self, other):
+        """
+        Should be used when ``other`` is not a sequence. Should be
+        defined to define custom behaviour.
+
+        Examples
+        ========
+
+        >>> from sympy import S, oo, SeqFormula
+        >>> from sympy.abc import n
+
+        >>> SeqFormula(n**2).coeff_mul(2)
+        SeqFormula((2*n**2, n), ([0, oo), 1))
+
+        Notes
+        =====
+
+        '*' defines multiplication of sequences with sequences only.
+        For multiplying sequences use ``mul`` method.
+        """
+        return Mul(self, coeff)
+
+    def _add(self, other):
+        """
+        Should only be used internally
+
+        self._add(other) returns a new, term-wise added sequence if self
+        knows how to add with other, otherwise it returns ``None``.
+
+        ``other`` should only be a sequence object.
+
+        Used within :class:`SeqAdd` class
+        """
+        return None
+
+    def _mul(self, other):
+        """
+        Should only be used internally
+
+        self._mul(other) returns a new, term-wise multiplied sequence if self
+        knows how to multiply with other, otherwise it returns ``None``.
+
+        ``other`` should only be a sequence object.
+
+        Used within :class:`SeqMul` class
+        """
+        return None
+
+    def __add__(self, other):
+        return self.add(other)
+
+    def __neg__(self):
+        return self.coeff_mul(-1)
+
+    def __sub__(self, other):
+        return self.sub(other)
+
+    def __mul__(self, other):
+        return self.mul(other)
+
+    def __iter__(self):
+        i = 0
+        while(i < self.length):
+            pt = self._ith_point(i)
+            yield self.coeff(pt)
+            i += 1
```

Comment:
This will not work. As length can be oo. So it will cause issues. Just found out.